tools/spanner/purge_ttl/src/main.rs

```diff
@@ -0,0 +1,49 @@
+use std::env;
+use std::error::Error;
+use std::sync::Arc;
+
+use googleapis_raw::spanner::v1::{
+    spanner::{CreateSessionRequest, ExecuteSqlRequest },
+    spanner_grpc::SpannerClient,
+};
+use grpcio::{CallOption, ChannelBuilder, ChannelCredentials, EnvBuilder, MetadataBuilder};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let url = env::var("SYNC_DATABASE_URL")?;
+    if !url.starts_with("spanner://") {
+        return Err("Invalid SYNC_DAYABASE_URL".into())
+    }
+
+    let database = url["spanner://".len()..].to_owned();
+    let endpoint = "spanner.googleapis.com:443";
+
+    // Set up the gRPC environment.
+    let env = Arc::new(EnvBuilder::new().build());
+    let creds = ChannelCredentials::google_default_credentials()?;
+
+    // Create a Spanner client.
+    let chan = ChannelBuilder::new(env.clone())
+        .max_send_message_len(100 << 20)
+        .max_receive_message_len(100 << 20)
+        .secure_connect(&endpoint, creds);
+    let client = SpannerClient::new(chan);
+
+    let mut req = CreateSessionRequest::new();
+    req.set_database(database.to_string());
+    let mut meta = MetadataBuilder::new();
+    meta.add_str("google-cloud-resource-prefix", &database)?;
+    meta.add_str("x-goog-api-client", "googleapis-rs")?;
+    let opt = CallOption::default().headers(meta.build());
+    let session = client.create_session_opt(&req, opt)?;
+
+    let mut req = ExecuteSqlRequest::new();
+    req.session = session.get_name().to_string();
+
+    req.sql = "DELETE FROM batches WHERE expiry < CURRENT_TIMESTAMP()".to_string();
+    client.execute_sql(&req)?;
```

Comment:
One thing we'll need to to execute this as a [Spanner "partitioned dml"](https://cloud.google.com/spanner/docs/dml-partitioned) statement.

The python client lib provides an easy to use `execute_partitioned_dml` function whereas our rust client lib doesn't (as it's a lower level library).

We can mimic what the python client does:

https://github.com/googleapis/python-spanner/blob/master/google/cloud/spanner_v1/database.py#L395

Basically: 

- Begin a transaction with the `partitioned_dml` TransactionOption
- Call execute_sql within that transaction

We have examples of creating transactions in the spanner/models.rs and the utilizing them w/ execute_sql calls in spanner/support.rs

Reply:
Great. Thanks for pointing this out.